src/plugins/vis_type_xy/public/vis_renderer.tsx

```diff
@@ -45,9 +46,10 @@ export const xyVisRenderer: ExpressionRenderDefinition<RenderValue> = {
   name: visName,
   displayName: 'XY visualization',
   reuseDomNode: true,
-  render: (domNode, { visData, visConfig, visType }, handlers) => {
+  render: async (domNode, { visData, visConfig, visType, syncColors }, handlers) => {
     const showNoResult = shouldShowNoResultsMessage(visData, visType);
-    const isSplitChart = Boolean(visConfig.dimensions.splitRow || visConfig.dimensions.splitRow);
+    const isSplitChart = Boolean(visConfig.dimensions.splitRow);
+    const palettes = await getPalettesService().getPalettes();
```

Comment:
Would this cause a delayed render if this request takes too long? I don't notice any delay myself but just a thought.

Reply:
@nickofthyme I thought about this and maybe it would be better to move it inside the component that is lazy loaded. You can see the change [here](https://github.com/elastic/kibana/pull/86876/commits/61bb3eabc96e5d79356b84f590801d92c703ab56), I think is better now so thanx ❤️